File: ABC/ABC337/D/main.cc
#include <algorithm>
#include <iostream>
#include <numeric>
#include <vector>

using namespace std;

template<class T>
vector<vector<T> > Transpose(const vector<vector<T> > &v) {
  int h = v.size(), w = v[0].size();
  vector ret(w, vector<T>(h));
  for (int r = 0; r < h; ++r) {
    for (int c = 0; c < w; ++c) {
      ret[c][r] = v[r][c];
    }
  }
  return ret;
}

template<class T>
vector<T> SlideSum(const vector<T> &v, int k) {
  vector<T> ret;
  if ((int)v.size() >= k) {
    T cur = accumulate(v.begin(), v.begin() + k, T{0});
    ret.emplace_back(cur);
    for (int i = k; i < (int)v.size(); ++i) {
      cur += v[i] - v[i - k];
      ret.emplace_back(cur);
    }
  }
  return ret;
}

template<class T>
T Solve(const vector<vector<T> > &vv, int k) {
  T ret = numeric_limits<T>::max();
  for (const auto &v : vv) {
    auto ss = SlideSum(v, k);
    if (!ss.empty()) {
      ret = min(ret, *min_element(ss.begin(), ss.end()));
    }
  }
  return ret;
}

int main() {
  int h, w, k;
  cin >> h >> w >> k;
  const long long INF = k + 1;

  vector s(h, vector<long long>(w, 0));
  for (int i = 0; i < h; ++i) {
    string str;
    cin >> str;
    for (int j = 0; j < w; ++j) {
      if (str[j] == 'x') {
        s[i][j] = INF;
      } else if (str[j] == '.') {
        s[i][j] = 1;
      }
    }
  }

  long long res = min(Solve(s, k), Solve(Transpose(s), k));
  cout << (res < INF ? res : -1) << endl;

  return 0;
}

File: ABC/ABC337/D/main.py
def transpose(li):
    h = len(li)
    w = len(li[0])
    ret = [[0] * h for _ in range(w)]
    for i in range(h):
        for j in range(w):
            ret[j][i] = li[i][j]
    return ret

def slide_sum(li, k):
    if len(li) < k:
        return [1e30]
    cur = 0
    for i in range(k):
        cur += li[i]
    ret = [cur]
    for i in range(k, len(li)):
        cur += li[i] - li[i - k]
        ret.append(cur)
    return ret

def solve(s, k):
    ret = k + 1
    for si in s:
        ss = slide_sum(si, k)
        ret = min(ret, min(ss))
    return ret

h, w, k = map(int, input().split())

inf = k + 1

s = [[0] * w for _ in range(h)]
for i in range(h):
    str = input()
    for j in range(w):
        if str[j] == 'x':
            s[i][j] = inf
        elif str[j] == '.':
            s[i][j] = 1

res = min(solve(s, k), solve(transpose(s), k))
print(res if res < inf else '-1')
